### ISSUE TITLE
Support for Google AI API integration?

### ISSUE THREAD

**Issue by ymma98** [archibate/genius.nvim#1]
A pleasure to see this brilliant vim plugin. I am just curious if there is any plan to support the API of Google AI (Gemini/Bard), which would extend the versatility of the plugin. 
Thank you for considering this asking.
Best regards!

**Comment by archibate** [comment]
I have recently tried out Google's genimi-pro model via Google AI Studio. However, it seems that Google automatically ignores all my input starting from `<`...
![图片](https://github.com/archibate/genius.nvim/assets/20640597/1d5cf291-f109-4a87-8f51-6f000258076d)
As you see in the picture, I used to have some good code examples here. But once I save and come back again, it automatically rids all my contents after `<`. Also the model seems cannot output `<` at all, maybe Google consider it as HTML tag which could be XSS attack?
Also, chat models are really bad at text insertion or completion. If Google have an working 'insertion' model (like gpt-3.5-turbo-instruct), rather than 'chat' model, I would be happy to integrate it into genius.nvim :)
